src/Campaigns/Actions/CampaignPageTemplate.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Framework\Views\View;
+
+/**
+ * @unreleased
+ */
+class CampaignPageTemplate
+{
+    /**
+     * @unreleased
+     */
+    public function registerTemplate()
+    {
+        if ( ! $this->canRegisterBlockTemplate()) {
+            return;
+        }
+
+        register_block_template('givewp//campaign-page-template', [
+            'title' => __('Campaign page', 'give'),
+            'description' => __('Give Campaign Page template', 'give'),
+            'post_types' => [
+                'give_campaign_page'
+            ],
+            'content' => View::load('Campaigns.campaign-page-content'),
+        ]);
+    }
+
+    /**
+     * @unreleased
+     */
+    public function loadTemplate($template)
+    {
+        if (
+            'give_campaign_page' === get_query_var('post_type')
+            && current_theme_supports('block-templates')
+        ) {
+            $template = $this->canRegisterBlockTemplate()
+                ? $template
+                : GIVE_PLUGIN_DIR . '/src/Campaigns/resources/views/campaign-page-template.php';
+
+            return locate_block_template($template, 'campaign-page-template', ['campaign-page-template.php']);
+        }
+
+        return $template;
+    }
```

Comment:
When hooking into `template_include`, we should first check if the user’s theme provides a template before falling back to our own. In this case, `single-give_campaign_page.php` should be prioritized if it exists in the theme.

Reply:
The whole purpose of this function is to load our template for themes that support FSE because they load a ton of stuff that we don't want to see on our campaign page. That's the point of this PR. Users can always change the layout from the editor if they don't like it. For sites that don't support FSE, the dev can create `single-give_campaign_page.php` and it will be loaded normally.

Reply:
![CleanShot 2025-02-28 at 13 00 56](https://github.com/user-attachments/assets/a6d2ef60-8e7e-4111-b36d-d2354dc50ffd)

This is an example of a theme without support for FSE. I thought that campaign-page-template.php would surface exactly in cases like that. It sounds like we are implementing it partially, so we’ll need to communicate that clearly to our users.